packages/h5wasm/src/guards.ts

```diff
@@ -93,3 +95,7 @@ export function assertNumericMetadata(
     throw new Error('Expected H5Wasm numeric metadata');
   }
 }
+
+export function hasInt64Type(dataset: Dataset) {
+  return dataset.type.class === DTypeClass.Integer && dataset.type.size === 64;
+}
```

Comment:
I could put this with other guards in `shared` but it is only useful for h5wasm and we don't need to assert the TS type like the guards in `shared` do.